### ISSUE TITLE
An error was reported when loading a custom dictionary.

### ISSUE THREAD

**Issue by tyf2018** [tadashi-aikawa/obsidian-various-complements-plugin#15]
An error was reported when loading a custom dictionary.
Are there any special restrictions on the size or encoding of dictionary files?

![2021-11-19_11-56-13](https://user-images.githubusercontent.com/56342788/142562361-86860ada-0ec9-4460-9ff6-8ed44c90d4db.jpg)

Best Regards.

**Comment by tyf2018** [comment]
There is indeed a limit to the size of the dictionary file, I divided it into multiple files of appropriate size, and it can be used normally.

Super awesome, thank you very much for your development!

**Comment by tadashi-aikawa** [comment]
@tyf2018 
Thank you for your bug report!

Would you tell me the size of the file that occurred error? I use Obsidian API to read local files, so I want to know the limit of size.

**Comment by tyf2018** [comment]
> Would you tell me the size of the file that occurred error? I use Obsidian API to read local files, so I want to know the limit of size.

The result of my own verification is that the dictionary file should not be larger than 1.3Mb. If the file is very large, it is recommended to divide it into multiple small files. At present, I have divided about 110 dictionary files, which are in good use.

The above data is likely to be inaccurate and is for reference only.

**Comment by tadashi-aikawa** [comment]
@tyf2018 
Thank you for the result of your verification. It is very helpful!

> I have divided about 110 dictionary files,

Oh.. it is too large more than expected 😅 

I think it is better that `custom dictionary paths` permits such a regexp pattern, do you think?

**Comment by tyf2018** [comment]
> I think it is better that permits such a regexp pattern, do you think?`custom dictionary paths`

Do you mean to let custom dictionary paths support regular expressions? This function is not that important to me. In fact, I don't know the regularity well and only use a few simple rules. But for those who understand regularity, it should be needed.

**Comment by tadashi-aikawa** [comment]
> Do you mean to let custom dictionary paths support regular expressions?

Yes, or glob patterns. If it is not essential for you, I will not add it, for now, thanks!

**Comment by tadashi-aikawa** [comment]
This is a bug in my code.. it occurs a Maximum call stack size exceeded error 😅

```typescript
this.customTokens.push(
  ...str.split(/(\r\n|\n)/).filter((x) => x !== "")
);
```

**Comment by tadashi-aikawa** [comment]
@tyf2018 
I have released [v1.2.1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/1.2.1).

Please try it and inform me it is working correctly or not :)

**Comment by tyf2018** [comment]
> I have released [v1.2.1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/1.2.1).
> 
> Please try it and inform me it is working correctly or not :)

Great, the new version has no restrictions on the size of dictionary files. My 20Mb file is used normally.
